File: api/src/models/Establishment.ts
import {Document, model, Model, Schema} from "mongoose";

const EstablishmentSchema: Schema = new Schema({
    name: {
        type: String,
        require: true,
    },
    establishment_type: {
        type: Schema.Types.ObjectId,
        ref: "establishmentTypes",
    },
    address: {
        number: {
            type: String,
            required: true,
        },
        street: {
            type: String,
            required: true,
        },
        zipCode: {
            type: String,
            required: true,
        },
        city: {
            type: String,
            required: true,
        },
        coordinate: {
            latitude: {
                type: Number,
                required: true,
            },
            longitude: {
                type: Number,
                required: true,
            },
        },
    },
});
interface IEstablishment {
    name: string;
    establishment_type: Schema.Types.ObjectId;
    address: {
        number: string;
        street: string;
        zipCode: string;
        city: string;
        coordinate: {
            latitude: string;
            longitude: number;
        };
    };
}
type EstablishmentType = IEstablishment & Document;
const EstablishmentModel: Model<EstablishmentType> = model(
    "establishments",
    EstablishmentSchema
);
export { EstablishmentModel, IEstablishment, EstablishmentType };

File: app/src/main/java/com/example/myhelloworld/model/Establishment.kt
package com.example.myhelloworld.model

import com.example.myhelloworld.map.models.Address

class Establishment {
    /**
     * Establishment name
     */
    var name: String

    /**
     * Establishment location
     */
    var address: Address

    /**
     * Type of this establishment
     */
    var establishment_type: EstablishmentType

    constructor(name: String, address: Address, type: EstablishmentType) {
        this.name = name
        this.address = address
        this.establishment_type = type
    }

}